src/KafkaFlow/Event.cs

```diff
@@ -0,0 +1,63 @@
+﻿namespace KafkaFlow
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    internal class Event<TArg> : IEvent<TArg>
+    {
+        private readonly ILogHandler logHandler;
+
+        private readonly IList<Func<TArg, Task>> handlers = new List<Func<TArg, Task>>();
+
+        public Event(ILogHandler logHandler)
+        {
+            this.logHandler = logHandler;
+        }
+
+        public void Subscribe(Func<TArg, Task> handler)
+        {
+            this.handlers.Add(handler);
```

Comment:
Should it be a requirement that if the same handler is registered more than once, the handler is called only one time?

The handler would only be registered if it hasn't already been added to the collection.

Although it wouldn't make much sense to register the same handler more than once, the user could do it by mistake. Should we add some protection around that?

Reply:
The method receives a function as handler, so it is difficult to validate if the content of the function was already added to the list.
But IMHO it should be managed by the users, this type of errors can happen in any event based implementation.